connector/src/main/java/org/geysermc/connector/entity/living/monster/EnderDragonEntity.java

```diff
@@ -61,31 +65,52 @@
     private final Segment[] segmentHistory = new Segment[19];
     private int latestSegment = -1;
 
-    private boolean hovering;
+    private int phase;
+    /**
+     * The number of ticks since the beginning of the phase
+     */
+    private int phaseTicks;
+
+    private int ticksTillNextGrowl = 100;
+    private int ticksSinceDeath = 0;
+
+    /**
+     * Used to determine when the wing flap sound should be played
+     */
+    private float wingPosition;
+    private float lastWingPosition;
 
-    private ScheduledFuture<?> partPositionUpdater;
+    private ScheduledFuture<?> entityTick;
 
     public EnderDragonEntity(long entityId, long geyserId, EntityType entityType, Vector3f position, Vector3f motion, Vector3f rotation) {
         super(entityId, geyserId, entityType, position, motion, rotation);
 
         metadata.getFlags().setFlag(EntityFlag.FIRE_IMMUNE, true);
+        metadata.putFloat(EntityData.HEALTH, 10); // So the dragon isn't considered dead before it receives health metadata
     }
 
     @Override
     public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession session) {
         // Phase
         if (entityMetadata.getId() == 15) {
-            int value = (int) entityMetadata.getValue();
-            if (value == 5) {
+            phase = (int) entityMetadata.getValue();
+            if (phase == 5) {
                 // Performing breath attack
                 EntityEventPacket entityEventPacket = new EntityEventPacket();
                 entityEventPacket.setType(EntityEventType.DRAGON_FLAMING);
                 entityEventPacket.setRuntimeEntityId(geyserId);
                 entityEventPacket.setData(0);
                 session.sendUpstreamPacket(entityEventPacket);
+            } else if (phase == 9) {
+                // Dying phase
+                EntityEventPacket entityEventPacket = new EntityEventPacket();
+                entityEventPacket.setType(EntityEventType.ENDER_DRAGON_DEATH);
+                entityEventPacket.setRuntimeEntityId(geyserId);
+                entityEventPacket.setData(0);
+                session.sendUpstreamPacket(entityEventPacket);
             }
-            metadata.getFlags().setFlag(EntityFlag.SITTING, value == 5 || value == 6 || value == 7);
-            hovering = value == 10;
+            phaseTicks = 0;
+            metadata.getFlags().setFlag(EntityFlag.SITTING, isSitting());
         }
         super.updateBedrockMetadata(entityMetadata, session);
     }
```

Comment:
I can't go lower than this because thanks GitHub - on line 131 we have:

`addEntityPacket.getAttributes().add(new AttributeData("minecraft:health", 0.0f, 200f, 200f, 200f));`

Is this still necessary with your new HEALTH metadata addition? If so, you can remove the entire spawn entity declaration and debug statement and just call super.